### DISCUSSION TITLE
Idea for realizing JMP <address>

### DISCUSSION THREAD

**Discussion by Odie-123** [nicolacimmino/PLC-14500#13] [Ideas]
I think it's possible to easily realize a JMP <addr> instruction by using two 4 bit parallel in/out latches. You also need the NOP0 and NOPF instruction. It works as follows:

1. `NOP0 <jmp-addr-4-lower-bits>` -> Stores the upper 4 data bits in latch 1
2. `NOPF <jmp-addr-4-higher-bits>` -> Stores the upper 4 data bits in latch 2
3. `JMP` -> Copies latch 1 + 2 into the program counter

NOP0 and NOPF can still be used as no-operation instruction, since it only stores a value in the latch. The upper 4 data bits of the JMP instruction could even be used for bank switching if you have more than 256 bytes of memory. The RTN instruction could instead be wired to the RESET line.

**Comment by nicolacimmino** [comment]
Yeah, that sounds like something that would work. I have no plan of reworking the PLC14500-Nano and, for a trainer board, I think the current features set is good and not too overwhelming for a newby.

I do have a half-baked plan to make another, more full computer like board based on the MC14500 and this idea could actually prove useful to avoid just increasing the RAM width (which was my first plan) or forcing the JMP to be aligned to a 256-byte page (which was my backup plan).

**Comment by Odie-123** [comment]
Yes I think the PLC14500 board is good the way it is. I like all those LEDs, they make a good show and you can see what's going on. I also built this board [TD4-4BIT-CPU](https://github.com/wuxx/TD4-4BIT-CPU) which is also a very nice design but there are only 4 LEDs and at first I thought it was defect because nothing happened until I entered a proper program.

I like the MC14500 and I also thought about how to make a full computer with it (with JMP and RTN). Instead of adding graphics you could add a RS232 interface to a TTY Terminal which could be controlled via ASCII escape sequences.

**Comment by Odie-123** [comment]
Do you know this 1-bit cpu design? It doesn't need a CPU, only an EPROM: [LAUGHTON ELECTRONICS One-bit Computing at 60 Hertz](https://laughtonelectronics.com/Arcana/One-bit%20computer/One-bit%20computer.html)

**Comment by nicolacimmino** [comment]
yes! I think I've seen it on Hackaday.
